packages/block-library/src/embed/embed-preview.js

```diff
@@ -59,7 +58,7 @@ class EmbedPreview extends Component {
 		const { interactive } = this.state;
 
 		const html = 'photo' === type ? getPhotoHtml( preview ) : preview.html;
-		const parsedHost = parse( url ).host.split( '.' );
+		const parsedHost = new URL( url ).host.split( '.' );
```

Comment:
Note that this will fail with an exception for scheme-relative URLs (`//example.com/path`), if it's possible for `url` to be one of those. If you want to support scheme-relative URLs, you'll need to supply a base URL. Here's one way of doing it (apologies if I made a mistake):

```js
const BASE_DOMAIN = '__domain__.invalid';
const BASE_URL = `https://${ BASE_DOMAIN }`;

const parsedUrl = new URL( url, BASE_URL );
const parsedHost = parsedUrl.hostname !== BASE_DOMAIN ? parsedUrl.hostname.split( '.' ) : [];
```

Also note that `parsedUrl.host` and `parsedUrl.hostname` are not the same; `.host` will include the port if provided, for example:

```js
const url = new URL( 'http://example.com:8080/path' );
url.host      // -> "example.com:8080"
url.hostname  // -> "example.com"
```

Finally, do bear in mind that this will throw an exception for invalid URLs, even if a base URL is provided. If you want to handle that situation, you may need a `try`/`catch`.

```js
new URL( 'http://example.com:badport/path', 'http://example.com' );
// Uncaught TypeError: Failed to construct 'URL': Invalid URL
```

Reply:
Yeah, the logic here definitely caught my attention as being potentially problematic. I actually started refactoring it a bit, and then realized that I had done this once more in the past when starting down a similar path of refactoring the code, and it proved to be quite a distraction to the original objective. For the purposes of this pull request, my primary goal is to implement a direct port of the equivalent behavior using `URL`.

So, in response, I'd consider: Do you think it's any more problematic than it had been previously when it was using `require( 'url' ).parse` ?

The way I thought about it: `URL` can throw when it encounters a URL that it cannot parse (which is not good). `parse` is a little more tolerant but only in that it won't throw an error. `require('url').parse('//example.com/path').host` would have evaluated to `null` previously, so calling `null.split( '.' )` would have been throwing an error anyways.

I don't want to disregard the suggestion, because I'm very much in agreement with you. But if it's something we can follow-up on separately, I think it would help avoid slowing this one down.

Reply:
I was under the impression that `url.parse` could handle scheme-relative URLs, but trying it out just now that does not appear to be the case. That's surprising for me, since it can pretty much handle every other kind of URL, including path-absolute and path-relative.

With that in mind, I'm in complete agreement with you; your code maintains the overall behaviour of this method, if we ignore the error type being returned.

You may still want to consider using `hostname` instead of `host`, however, as `[ 'www', 'example', 'com:8080' ]` is probably not what you're looking for out of `'http://www.example.com:8080'`, I assume?

Reply:
> You may still want to consider using `hostname` instead of `host`, however, as `[ 'www', 'example', 'com:8080' ]` is probably not what you're looking for out of `'http://www.example.com:8080'`, I assume?

I think you are correct and I agree with you, but I want to keep the scope of this pull request as constrained as possible, and I'd rather avoid to make any changes not strictly maintaining an identical behavior that would have existed previously. It can be difficult enough as it is to get an approving review 😅 

For practical consideration, I doubt there's real cases this is going to fail. This is handling links copied from Facebook and SmugMug. I don't think we would anticipate any port-ed URLs.

To be honest, I'd rather just get rid of this altogether. I have an old pull request at #13755 and specifically a remark at https://github.com/WordPress/gutenberg/pull/13755/files#r254896101 where I have some prior agenda toward this. I might like to follow-up to refresh it.